src/app/components/question-approval-page/question-approval-page.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component, OnInit } from '@angular/core';
+import { QuestionDto } from 'src/app/models/question-dto';
+import { QuestionService } from 'src/app/services/question.service';
+import { FormControl, FormGroup } from '@angular/forms';
+@Component({
+  selector: 'app-question-approval-page',
+  templateUrl: './question-approval-page.component.html',
+  styleUrls: ['./question-approval-page.component.scss']
+})
+export class QuestionApprovalPageComponent implements OnInit {
+  question: QuestionDto =  QuestionDto.createDtoWithNonEmptyFields();
+  exampleForm = new FormGroup({
+    sample: new FormControl(''),
+  });
+  constructor(
+    private questionService: QuestionService,
+  ) { }
+// maybe add put back button for concurrency issues. also add checks against approval 
+  ngOnInit(): void {
+
+    this.questionService.getQuestionFromQueue().
+    subscribe(queue => this.question = new QuestionDto(JSON.parse(queue.messageText)));
```

Comment:
```suggestion
this.question = this.questionService.getQuestionFromQueue().pipe(
      map(questionFromQueue =>new QuestionDto(JSON.parse(queue.messageText)))
      );
```

Reply:
I'll bring this up later so I can understand the differences before I make this change